### ISSUE TITLE
[no-unused-vars] false positive for Type Import 

### ISSUE THREAD

**Issue by mvnrhd** [typescript-eslint/typescript-eslint#2946]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
   "@typescript-eslint/no-unused-vars": [
            "error",
            {
                "argsIgnorePattern": "^_"
            }
        ]
  }
}
```

```TS
import { ElementRef } from '@angular/core';
...
@ViewChild('someInput', { static: false })
private readonly _someInput?: ElementRef<HTMLInputElement>;
```

**Expected Result**
No error

**Actual Result**
 `error    'ElementRef' is defined but never used           @typescript-eslint/no-unused-vars`

**Additional Info**

Occurs after latest update to 4.14.0 (probably related to [#2751](https://github.com/typescript-eslint/typescript-eslint/pull/2751)?)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.14.0` |
| `@typescript-eslint/parser`        | `4.14.0` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.18.0` |
| `node`                             | `14.15.4` |

**Comment by miluoshi** [comment]
Also the same issue mentioned here: https://github.com/typescript-eslint/typescript-eslint/pull/2751#issuecomment-762787540

**Comment by bradzacher** [comment]
Same as #2941 and #2942, and fixed in 

https://github.com/typescript-eslint/typescript-eslint/pull/2943
